File: spring-boot-oauth2-auth-server/src/main/java/com/bittokazi/oauth2/auth/server/app/services/user/helpers/UserHelpers.java
package com.bittokazi.oauth2.auth.server.app.services.user.helpers;

import com.bittokazi.oauth2.auth.server.app.models.tenant.User;
import com.bittokazi.oauth2.auth.server.app.repositories.tenant.UserRepository;
import com.bittokazi.oauth2.auth.server.utils.Utils;
import org.apache.coyote.Constants;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Sort;
import org.springframework.data.domain.Sort.Direction;

import java.security.NoSuchAlgorithmException;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.UUID;
import java.util.stream.Collectors;

/**
 * @author Bitto Kazi
 */

public class UserHelpers {

	public static Object getUsers(int page, int count, UserRepository userRepository) {
		Map<String, Object> json = new HashMap<>();
		Pageable reqCount = PageRequest.of(page, count, Sort.by(Direction.DESC, "id"));
		Page<User> pages = userRepository.findAll(reqCount);
		json.put("users", UserHelpers.setUsersImage(pages.getContent()));
		json.put("pages", pages.getTotalPages());
		json.put("records", pages.getTotalElements());
		return json;
	}

	public static User setUserImage(User user) {
		try {
			user.setAvatarImage("https://www.gravatar.com/avatar/" + Utils.getMD5(user.getEmail()) + "?d=identicon");
		} catch (NoSuchAlgorithmException e) {
			e.printStackTrace();
		}
		return user;
	}

	public static List<User> setUsersImage(List<User> users) {
		return users.stream().map(user -> {
			return UserHelpers.setUserImage(user);
		}).collect(Collectors.toList());
	}

}

File: spring-boot-oauth2-auth-server/src/main/kotlin/com/bittokazi/oauth2/auth/server/app/services/user/helpers/UserHelpers.kt
package com.bittokazi.oauth2.auth.server.app.services.user.helpers

import com.bittokazi.oauth2.auth.server.app.models.tenant.User
import com.bittokazi.oauth2.auth.server.app.models.tenant.UserList
import com.bittokazi.oauth2.auth.server.app.repositories.tenant.UserRepository
import com.bittokazi.oauth2.auth.server.utils.Utils.getMD5
import org.springframework.data.domain.PageRequest
import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Sort
import java.security.NoSuchAlgorithmException
import java.util.stream.Collectors

/**
 * @author Bitto Kazi
 */
object UserHelpers {
    fun getUsers(page: Int, count: Int, userRepository: UserRepository): Any {
        val reqCount: Pageable = PageRequest.of(page, count, Sort.by(Sort.Direction.DESC, "createdDate"))
        val pages = userRepository.findAll(reqCount)
        return UserList(
            pages = pages.totalPages,
            records = pages.totalElements,
            users = setUsersImage(pages.content)
        )
    }

    fun setUserImage(user: User?): User? {
        try {
            if (user != null) {
                user.avatarImage = "https://www.gravatar.com/avatar/" + getMD5(
                    user!!.email!!
                ) + "?d=identicon"
            }
        } catch (e: NoSuchAlgorithmException) {
            e.printStackTrace()
        }
        return user
    }

    fun setUsersImage(users: MutableList<User?>): List<User?> {
        return users.stream().map { user: User? -> setUserImage(user) }
            .collect(Collectors.toList())
    }
}
